module.nix

```diff
@@ -171,9 +195,9 @@ in
             # TODO maybe make this configurable
             ExecStart = "${cfg.authentikComponents.celery}/bin/celery -A authentik.root.celery worker -Ofair --max-tasks-per-child=1 --autoscale 3,1 -E -B -s /tmp/celerybeat-schedule -Q authentik,authentik_scheduled,authentik_events";
             EnvironmentFile = mkIf (cfg.environmentFile != null) [ cfg.environmentFile ];
-            LoadCredential = mkIf (cfg.nginx.enable && cfg.nginx.enableACME) [
-              "${cfg.nginx.host}.pem:${config.security.acme.certs.${cfg.nginx.host}.directory}/fullchain.pem"
-              "${cfg.nginx.host}.key:${config.security.acme.certs.${cfg.nginx.host}.directory}/key.pem"
+            LoadCredential = mkIf (cfg.nginx.enable && (cfg.nginx.vhost.enableACME || cfg.nginx.vhost.useACMEHost != null)) [
+              "${cfg.nginx.vhost.useACMEHost ? cfg.nginx.host}.pem:${config.security.acme.certs.${cfg.nginx.vhost.useACMEHost ? cfg.nginx.host}.directory}/fullchain.pem"
```

Comment:
:joy: half way there, please do one thing: try to evaluate this when you set `enableACME` instead of `useACMEHost`, i know it is not your usecase, but it is mine, and currently you still break it.

Reply:
should now be correct, but I am still testing

Reply:
Thanks.